authCommon/app/modules/dao/AllowedIssuerDao.js

```diff
@@ -149,11 +191,28 @@ function prepareTransactionInput(cfg, downloadUrl, jwksBinaryBody, modificationT
 
 async function addJwksCacheEntry(iss, downloadUrlFn){
     const cfg = await getConfigByISS(iss)
-
-    const jwksBodyByteArray = await downloadUrlFn(cfg.JWKSUrl)
+    const dynamoCacheLimit = process.env.JWKS_DYNAMO_CACHE_CONTENT_LIMIT
+    const jwksBodyBinary = await downloadUrlFn(cfg.JWKSUrl)
+    const jwksBinaryBodySize = Buffer.from(jwksBodyBinary).length
     const modificationTimeEpochMs = Date.now()
+    let jwksS3Url;
+
+    if(jwksBinaryBodySize > dynamoCacheLimit) {
+        const bucketName = process.env.JWKS_CONTENTS
+        const sha256 = computeSha256(jwksBodyBinary)
+        const cacheMaxUsageEpochSec = Math.floor( modificationTimeEpochMs / 1000) + cfg.JWKSCacheMaxDurationSec
+        const input = preparePutObjectInput(bucketName, iss, sha256, cfg.JWKSUrl, jwksBodyBinary, cacheMaxUsageEpochSec)
+        try {
+            await putObject(input)
+        }
+        catch (error) {
+            console.warn(error)
+            throw new Error("Error uploading S3 object " + cfg.JWKSUrl)
+        }
+        jwksS3Url = 's3://' + bucketName + "/" + prepareKeyInput(iss, cfg.JWKSUrl, sha256)
```

Comment:
@ViVeniero 
- spostare il codice da riga 201.a riga 212 in una funzione dedicata
- spostare la generazione di cacheMaxUsageEpochSec in una funzione dedicata visto che ora è duplicato sia qui che in prepareTransactionInput